packages/react-badge/src/components/PresenceBadge/usePresenceBadge.tsx

```diff
@@ -0,0 +1,55 @@
+import * as React from 'react';
+import { makeMergeProps, ObjectShorthandProps } from '@fluentui/react-utilities';
+import { PresenceBadgeProps, PresenceBadgeState, PresenceBadgeStatus } from './PresenceBadge.types';
+import { useBadge, BadgeProps } from '../Badge/index';
+import {
+  SkypeMinusIcon,
+  SkypeClockIcon,
+  SkypeArrowIcon,
+  SkypeCheckIcon,
+  CancelIcon,
+} from './DefaultPresenceBadgeIcons';
+
+/**
+ * Consts listing which props are shorthand props.
+ */
+export const presenceBadgeShorthandProps: (keyof PresenceBadgeProps)[] = ['icon'];
+
+const mergeProps = makeMergeProps<PresenceBadgeState>({ deepMerge: presenceBadgeShorthandProps });
+
+const iconMap: Record<PresenceBadgeStatus, JSX.Element> = {
+  busy: <SkypeMinusIcon />,
+  available: <SkypeCheckIcon />,
+  away: <SkypeClockIcon />,
+  offline: <CancelIcon />,
+  outOfOffice: <SkypeArrowIcon />,
+  doNotDisturb: <></>,
+};
+
+/**
+ * Returns the props and state required to render the component
+ */
+export const usePresenceBadge = (
+  props: PresenceBadgeProps,
+  ref: React.Ref<HTMLElement>,
+  defaultProps?: PresenceBadgeProps,
+): PresenceBadgeState => {
+  const state = useBadge(
+    props,
+    ref,
+    mergeProps(
+      {
+        size: 'small',
+        status: 'available',
+        outOfOffice: false,
+      },
```

Comment:
If you use my other suggestion about only checking `state.icon.children` below, you'd need to provide a default for the `icon` slot here. (Not sure if it should be a `div` or `span` by default?):
```suggestion
        icon: { as: 'div' },
      },
```